testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentContainers.java

```diff
@@ -45,16 +55,27 @@ public static boolean isPrestoContainer(String name)
     public static void configureTempto(Environment.Builder builder, ResourceProvider configDir)
     {
         builder.configureContainer(TESTS, dockerContainer -> {
-            String environmentName = configDir.getPath("..").toFile().getName();
-            String temptoConfig = "/docker/presto-product-tests/conf/tempto/tempto-configuration-for-" + environmentName + ".yaml";
+            Path path = configDir.getPath("tempto-configuration.yaml");
+            String suffix = getParentDirectoryName(path) + "-" + randomSuffix();
+            String temptoConfig = "/docker/presto-product-tests/conf/tempto/tempto-configuration-for-" + suffix + ".yaml";
             dockerContainer
-                    .withCopyFileToContainer(
-                            forHostPath(configDir.getPath("tempto-configuration.yaml")),
-                            temptoConfig)
+                    .withCopyFileToContainer(forHostPath(path), temptoConfig)
                     .withEnv("TEMPTO_CONFIG_FILES", temptoConfigFiles ->
                             temptoConfigFiles
                                     .map(files -> files + "," + temptoConfig)
                                     .orElse(temptoConfig));
         });
     }
+
+    private static String getParentDirectoryName(Path path)
+    {
+        checkArgument(path.getNameCount() > 2, "Parent directory name not found in: " + path);
```

Comment:
`>=`?

Reply:
`Cannot determine parent directory of ` + path ?

Reply:
> >=?

Yes, it has to have a parent directory. So we expect two names here: `/parent.name/file.name`

Reply:
Yeah - two. Not necessarily more than two.